### PR TITLE
Split ConsoleKit into two functional targets

### PR DESCRIPTION
**Pull request by gwynne**
ConsoleKit has been split into two separate targets:

- `ConsoleKitTerminal` contains the logic for interacting with a console (I/O, color support, other ANSI commands, etc.).
- `ConsoleKitCommands` contains the functionality for handling argument parsing, which is now soft-deprecated in favor of `swift-argument-parser` when possible.

The existing `ConsoleKit` target is now an umbrella import, similar to the function of the `NIO` target in `swift-nio`; existing code should be unaffected.

Several additional minor cleanups were also made in the process of this split.

### CODE REVIEW

File: Sources/ConsoleKitCommands/Async/AsyncCommandGroup.swift [hunk 1]
```diff
@@ -60,8 +62,8 @@ extension AsyncCommandGroup {
             context.console.print(self.help)
         }
 
-        let padding = self.commands.map { $0.key }.longestCount + 2
-        if self.commands.count > 0 {
+        let padding = (self.commands.max(by: { $0.key.count < $1.key.count })?.key.count ?? 0) + 2
```
Comment (MahdiBM) [inline Sources/ConsoleKitCommands/Async/AsyncCommandGroup.swift:65]:
```suggestion
        let padding = (self.commands.map(\.key.count).max() ?? 0) + 2
```

File: Sources/ConsoleKitCommands/CommandGroup.swift [hunk 2]
```diff
@@ -60,8 +62,8 @@ extension CommandGroup {
             context.console.print(self.help)
         }
 
-        let padding = self.commands.map { $0.key }.longestCount + 2
-        if self.commands.count > 0 {
+        let padding = (self.commands.max(by: { $0.key.count < $1.key.count })?.key.count ?? 0) + 2
```
Comment (MahdiBM) [inline Sources/ConsoleKitCommands/CommandGroup.swift:65]:
```suggestion
        let padding = (self.commands.map(\.key.count).max() ?? 0) + 2
```

File: Sources/ConsoleKitCommands/Command.swift [hunk 3]
```diff
@@ -138,8 +140,8 @@ extension CommandSignature {
             + self.arguments.map { $0.name }
             + self.flags.map { $0.name }
 
-        let padding = names.longestCount + 2
-        if self.arguments.count > 0 {
+        let padding = (names.max(by: { $0.count < $1.count })?.count ?? 0) + 2
```
Comment (MahdiBM) [inline Sources/ConsoleKitCommands/Command.swift:143]:
```suggestion
        let padding = (names.map(\.count).max() ?? 0) + 2
```

File: Sources/ConsoleKitCommands/Utilities.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+extension Array {
+    /// Pops the first element from the array.
+    mutating func popFirst() -> Element? {
+        guard let pop = first else {
+            return nil
+        }
+        self = Array(dropFirst())
+        return pop
+    }
+}
```
Comment (MahdiBM) [inline Sources/ConsoleKitCommands/Utilities.swift:13]:
```suggestion
extension Array {
    /// Pops the first element from the array.
    mutating func popFirst() -> Element? {
        guard let pop = self.first else {
            return nil
        }
        self = Array(self.dropFirst())
        return pop
    }
}
```

**Review by MahdiBM — CHANGES_REQUESTED** [review]

File: Sources/ConsoleKitCommands/Async/AsyncCommandGroup.swift (hunk 1 above)
Comment (MahdiBM) [reply Sources/ConsoleKitCommands/Async/AsyncCommandGroup.swift:65]: BTW this is a suggestion for performance, not a nit. It just happens to be cleaner too.

File: Sources/ConsoleKitAsyncExample/entrypoint.swift [hunk 5]
```diff
@@ -16,7 +16,6 @@ struct AsyncExample {
             try await console.run(group, input: input)
         } catch let error {
             console.error("\(error)")
-            exit(1)
```
Comment (0xTim) [inline Sources/ConsoleKitAsyncExample/entrypoint.swift:19]: Why not `exit(1)` anymore?

File: Sources/ConsoleKitCommands/Docs.docc/index.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# ``ConsoleKit``
```
Comment (0xTim) [inline Sources/ConsoleKitCommands/Docs.docc/index.md:1]:
```suggestion
# ``ConsoleKitCommands``
```

File: Sources/ConsoleKitCommands/Docs.docc/index.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+@Metadata {
+    @TitleHeading(Package)
+}
+
+ConsoleKit provides utilities for interacting with a console via a Swift application. It provides:
+
+* A ``Command`` type for writing commands with arguments and flags
+* Utilities for sending and receiving text to a terminal
+* A [Swift Log](https://github.com/apple/swift-log) implementation for a ``Logger`` that outputs to the console
```
Comment (0xTim) [inline Sources/ConsoleKitCommands/Docs.docc/index.md:11]: This lives in the other package

File: Sources/ConsoleKitTerminal/Docs.docc/index.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# ``ConsoleKit``
```
Comment (0xTim) [inline Sources/ConsoleKitTerminal/Docs.docc/index.md:1]:
```suggestion
# ``ConsoleKitTerminal``
```

File: Sources/ConsoleKitTerminal/Docs.docc/index.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+@Metadata {
+    @TitleHeading(Package)
+}
+
+ConsoleKit provides utilities for interacting with a console via a Swift application. It provides:
+
+* A ``Command`` type for writing commands with arguments and flags
```
Comment (0xTim) [inline Sources/ConsoleKitTerminal/Docs.docc/index.md:9]: Other package

**Review by 0xTim — APPROVED** [review]
Documentation needs updating otherwise mainly LGTM

**Review by ptoffy — APPROVED** [review]
Nothing to add to the other reviews

File: Sources/ConsoleKitAsyncExample/entrypoint.swift (hunk 5 above)
Comment (gwynne) [reply Sources/ConsoleKitAsyncExample/entrypoint.swift:19]: 'cause it had no meaningful effect, and exiting with a nonzero status is arguably a questionable behavior to provide as an example (especially when `exit(1)` isn't a fully platform-agnostic or even necessarily correct means of doing so)

**Pull request merged**